linker/Linker/LinkContext.cs

```diff
@@ -138,18 +140,17 @@ public LinkContext (Pipeline pipeline, AssemblyResolver resolver)
 			: this(pipeline, resolver, new ReaderParameters
 			{
 				AssemblyResolver = resolver
-			},
-			new AnnotationStore ())
+			})
 		{
 		}
 
-		public LinkContext (Pipeline pipeline, AssemblyResolver resolver, ReaderParameters readerParameters, AnnotationStore annotations)
+		public LinkContext (Pipeline pipeline, AssemblyResolver resolver, ReaderParameters readerParameters)
 		{
 			_pipeline = pipeline;
 			_resolver = resolver;
 			_actions = new Dictionary<string, AssemblyAction> ();
 			_parameters = new Dictionary<string, string> ();
-			_annotations = annotations;
+			_annotations = new AnnotationStore (this);
```

Comment:
Can you use the Create pattern I used in https://github.com/mono/linker/pull/208 ?

Our linker has our own AnnotationsStore that we need to create

Reply:
That's very bad practice, could you pass the `annotations` value as an argument instead ?

Reply:
I understand it's not great.  Why is it `very bad`?

We can (and do today) pass in AnnotationsStore as an argument to LinkContext constructor.  However, once AnnotationsStore needs LinkContext passed to it at instantiation you can no longer do that.  An alternative would be to expose a pubic property to set `annotations.Context = this` inside LinkContext .ctor().  However, I'd argue a public setter is worse

We could do a more invasive refactor to do something nicer, but is it worth the effort?

Reply:
I wasn't sure how you use the virtual methods, so I rather left it out from this PR. I can add it in another one.

Reply:
As long as a solution follows relatively soon after.  This is an API break for us.  I can't keep our linker in sync with upstream until I have some way to create our UnityAnnotationStore instead of the default AnnotationStore.  My virtual method recommendation is my "good enough suggestion".  If you want to go with a different approach, feel free.